### PR TITLE
feat(grouping): Always update grouping project configs

### PR DESCRIPTION
**Pull request by armenzg**
We don't want customers to be stuck in old project configs.

Related getsentry PR: https://github.com/getsentry/getsentry/pull/14685/files

### CODE REVIEW

File: src/sentry/conf/server.py [hunk 1]
```diff
@@ -2975,7 +2975,8 @@ def custom_parameter_sort(parameter: dict) -> tuple[str, int]:
 
 # If this is turned on, then sentry will perform automatic grouping updates.
 # This is enabled in production
-SENTRY_GROUPING_AUTO_UPDATE_ENABLED = False
+# XXX: Remove from getsentry first, then here
+SENTRY_GROUPING_AUTO_UPDATE_ENABLED = True
```
Comment (armenzg) [inline src/sentry/conf/server.py:2979]: We will need to follow up with another PR on `getsentry` once this one merges.

File: src/sentry/grouping/ingest/config.py [hunk 2]
```diff
@@ -17,24 +17,13 @@
...
-    should_update_org = (
-        project.organization_id % 1000 < float(settings.SENTRY_GROUPING_AUTO_UPDATE_ENABLED) * 1000
-    )
-    is_deprecated_config = project.get_option("sentry:grouping_config") in CONFIGS_TO_DEPRECATE
-    auto_update = bool(project.get_option("sentry:grouping_auto_update"))
-    return is_deprecated_config or (auto_update and should_update_org)
+# We want to test the new optimized transition code with projects that are on
+# deprecated configs before making it the default code path.
+# Remove this once we have switched to the optimized code path.
+DO_NOT_UPGRADE_YET = ("legacy:2019-03-12", "newstyle:2019-10-29", "newstyle:2019-05-08")
```
Comment (armenzg) [inline src/sentry/grouping/ingest/config.py:26]: This is our exclude list.

File: tests/sentry/event_manager/test_event_manager_grouping.py [hunk 3]
```diff
@@ -137,60 +136,48 @@ def test_updates_group_metadata(self):
     def test_auto_updates_grouping_config(self):
         self.project.update_option("sentry:grouping_config", LEGACY_CONFIG)
 
-        with override_settings(SENTRY_GROUPING_AUTO_UPDATE_ENABLED=False):
```
Comment (armenzg) [inline tests/sentry/event_manager/test_event_manager_grouping.py:140]: This test change can be more easily seen by hiding white spaces. This screenshot should save you having to reload the page:
image

File: tests/sentry/event_manager/test_event_manager_grouping.py [hunk 4]
```diff
@@ -137,60 +136,48 @@ def test_updates_group_metadata(self):
...
+        # ever-so-slightly before the audit log entry's timestamp, making a one-second tolerance
+        # necessary.
+        actual_expiry = audit_log_entry.data["sentry:secondary_grouping_expiry"]
+        expected_expiry = (
+            int(audit_log_entry.datetime.timestamp()) + SENTRY_GROUPING_UPDATE_MIGRATION_PHASE
+        )
+        assert actual_expiry == expected_expiry or actual_expiry == expected_expiry - 1
 
-    def test_disabled_auto_update_does_not_update(self):
+    def test_does_not_update_project_in_exclude_list(self):
```
Comment (armenzg) [inline tests/sentry/event_manager/test_event_manager_grouping.py:169]: Adding a new test for the exclude list. Once we have switched to the optimized code path we can remove it.

File: src/sentry/grouping/ingest/config.py [hunk 5]
```diff
@@ -17,24 +17,18 @@
 
 Job = MutableMapping[str, Any]
 
-# We are moving all projects off these configuration without waiting for events
-CONFIGS_TO_DEPRECATE: list[str] = [
-    "mobile:2021-02-12",
-]
+# Used to migrate projects that have no activity via getsentry scripts
+CONFIGS_TO_DEPRECATE = "mobile:2021-02-12"
```
Comment (lobsterkatie) [inline src/sentry/grouping/ingest/config.py:21]: Why switch this not to being a list (tuple, set, whatever)? There might be multiple in the future, no?

**Review by lobsterkatie — APPROVED** [review]

**Review by vartec — APPROVED** [review]
LGTM!

**Pull request merged**